xdsl/ir/core.py

```diff
@@ -417,10 +417,10 @@ def new(cls: type[Self], params: Any) -> Self:
         """
         # Create the new attribute object, without calling its __init__.
         # We do this to allow users to redefine their own __init__.
-        attr = cls.__new__(cls, params)
+        attr = cls.__new__(cls)
 
         # Call the __init__ of Data, which will set the parameters field.
-        Data[Any].__init__(attr, params)
+        Data.__init__(attr, params)  # type: ignore
```

Comment:
Okay this is confusing as hell, but basically, the *object* `Data[Any]` *is not* `Data`, even if the two types are the same, because `Data[Any]` is a `GenericTypeAlias` :upside_down_face: